### PR TITLE
Run Note Endpoint

### LINKED ISSUE #248: [Scylla] - Endpoint for changing a Run note
### Description

Right now we only have support for updating coordinates of a run, fetching runs based on ID, and creating runs. We want to have a endpoint in scylla that can update a Run note (run notes are for engineers to make notes while we collecting data, which are separated by Runs.

### Acceptance Criteria

Service is tested (in run_service_test.rs) and works as expected (updates a runs note). 
Route is tested using Postman, if you haven't used Postman before here is an explination by the Goat himself: [https://www.youtube.com/watch?v=CgFTb_hKdEo&feature=youtu.be](https://www.youtube.com/watch?v=CgFTb_hKdEo&feature=youtu.be) 

If you have any questions about Postman testing (after watching the video), let me know.

### Proposed Solution

1. Add a service in run_service.rs that takes the the db, run_id and note, and updates the note of the run with the given ID (see other services that use diesel::update as examples), that returns the updated Run
2.  Add a controller in run_controllers.rs that takes in that run id and a note through Path (see other functions as an example), and calls your service function and then puts it's output (Run) into a into the PublicRun transformer and returns the result or an error (see other functions as examples)
3. Add a route to the Router in main.rs (follow the format of other routes (except this time take in a note and a Run ID))., that uses your controller.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by justinsoto**
…run note

## Changes

_Explanation of changes goes here_

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `package-lock.json` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #248

### CODE REVIEW

File: scylla-server/src/controllers/run_controller.rs [hunk 1]
```diff
@@ -54,3 +54,38 @@ pub async fn new_run(State(pool): State<PoolHandle>) -> Result<Json<PublicRun>,
```
Comment (bracyw) [inline scylla-server/src/controllers/run_controller.rs:54]: Just run cargo fmt.... do get build checks to work. (you can check on your end with cargo fmt --check.

File: scylla-server/src/main.rs [hunk 2]
```diff
@@ -247,6 +247,11 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .route("/runs", get(run_controller::get_all_runs))
         .route("/runs/{id}", get(run_controller::get_run_by_id))
         .route("/runs/new", post(run_controller::new_run))
+        .route("/runs/new/{driver}/{location}/{notes}", post(run_controller::new_run_with_data))
+        .route(
+            "/runs/update/{id}/{notes}",
```
Comment (bracyw) [inline scylla-server/src/main.rs]: update should probably take driver and location as well... just in case we ever want to change that. I was thinking location could be used for something like testing in different areas... not just the general test day location. Unless Jack said otherwise.

**Review by bracyw — CHANGES_REQUESTED** [review]
Looks good just those small changes and I'll merge at meeting today.

**Review by bracyw — APPROVED** [review]
lgtm
